cmd/scheduler/app/options/options.go

```diff
@@ -47,24 +47,25 @@ const (
 
 // ServerOption is the main context object for the controller manager.
 type ServerOption struct {
-	KubeClientOptions    kube.ClientOptions
-	CertFile             string
-	KeyFile              string
-	CaCertFile           string
-	CertData             []byte
-	KeyData              []byte
-	CaCertData           []byte
-	SchedulerNames       []string
-	SchedulerConf        string
-	SchedulePeriod       time.Duration
-	EnableLeaderElection bool
-	LockObjectNamespace  string
-	DefaultQueue         string
-	PrintVersion         bool
-	EnableMetrics        bool
-	ListenAddress        string
-	EnablePriorityClass  bool
-	EnableCSIStorage     bool
+	KubeClientOptions                kube.ClientOptions
+	CertFile                         string
+	KeyFile                          string
+	CaCertFile                       string
+	CertData                         []byte
+	KeyData                          []byte
+	CaCertData                       []byte
+	SchedulerNames                   []string
+	SchedulerConf                    string
+	SchedulePeriod                   time.Duration
+	EnableLeaderElection             bool
+	LockObjectNamespace              string
+	DefaultQueue                     string
+	PrintVersion                     bool
+	EnableMetrics                    bool
+	ListenAddress                    string
+	EnablePriorityClass              bool
+	EnableCSIStorage                 bool
+	EnableDynamicResourcesAllocation bool
```

Comment:
We should set a separate feature struct like https://github.com/volcano-sh/volcano/blob/33a9eb5b96cddc3c076a439ed134d41a0d095220/pkg/scheduler/plugins/predicates/predicates.go#L291，which is imported from k8s instead of putting it in `ServerOption`, cause it's a plugin switch.